File: driftwood-e2e/src/pages/HomePage.js
const BasePage = require('./BasePage');

/**
 * Page Model for the HomePage
 */
class HomePage extends BasePage {
    /**
     * Construct the home page
     * @param browser the browser
     */
    constructor(browser) {
        super(browser);
    }

    /**
     * Open the home page
     */
    async visit() {
        this._browser.visit('/');
    }
}

module.exports = HomePage;

File: driftwood-e2e/src/main/kotlin/uk/co/grahamcox/driftwood/e2e/pages/HomePage.kt
package uk.co.grahamcox.driftwood.e2e.pages

import uk.co.grahamcox.driftwood.e2e.browser.Browser

/**
 * Representation of the Home Page
 */
class HomePage(browser: Browser) : BasePage(browser) {

    /**
     * Navigate to the page
     */
    fun navigateTo() {
        navigateTo("/")
    }
}
